### PR TITLE
Make SearchResult objects JSON.stringify'able

### PR DESCRIPTION
**Pull request by grncdr**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -139,21 +139,24 @@ var Field = redefine.Class({
...
 var SearchResult = redefine.Class({
-  constructor: function SearchResult() {},
+  constructor: function SearchResult(ItemType, object) {
+    this.responseData = object;
+    this.total = object.total;
+    this.limit = object.limit;
+    this.skip = object.skip;
+    walkMutate(_.cloneDeep(object), isParseableResource, parseResource);
+    resolveResponse(object).forEach(function (item, i) {
+      this[i] = item;
```
Comment (madtrick) [inline index.js]: why are you assigning in the SearchResult instance every item to a property named after its index?

Comment (grncdr) [inline index.js]: It's intended to be backward compatible with the current API (which returns an augmented array).

Comment (grncdr) [inline index.js]: Although actually that makes no sense at all so maybe I will try something else.

File: index.js [hunk 2]
```diff
@@ -138,24 +138,30 @@ var Field = redefine.Class({
...
-          enumerable: false
-        }
-      );
-    }
-  }
+SearchResult.prototype.toJSON = function () {
+  return this.responseData;
+};
+
+Object.keys(Array.prototype).filter(function (key) {
```
Comment (sdepold) [inline index.js]: Why are we doing this?

File: test/integration/content-type-test.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ buster.testCase('Content Type', {
       'order': 'name',
       'sys.id[in]': 'cat,dog,human'
     }).then(function(contentTypes) {
-      assert.match(contentTypes, [
+      assert.match(contentTypes.items, [
```
Comment (sdepold) [inline test/integration/content-type-test.js]: Why did this change? Isn't that a BC breaking change?

**Comment by madtrick** [comment]
@grncdr what's the current state of the PR?

File: test/integration/content-type-test.js (hunk 3 above)
Comment (grncdr) [inline test/integration/content-type-test.js]: yes, as commented in the commit message, this would break BC. The old API is not tenable.

**Comment by grncdr** [comment]
@madtrick - test failures need fixing and then we need to bump major version.

**Comment by grncdr** [comment]
This could be re-opened/worked on by anybody in the future, but I don't have the time to finish it now.

**Pull request closed without merging**